plugins/generic/externalFeed/controllers/grid/ExternalFeedGridHandler.inc.php

```diff
@@ -0,0 +1,240 @@
+		$this->setGridDataElements($feeds);
+		
+		// Add grid-level actions
+		$router = $request->getRouter();
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		$this->addAction(
+			new LinkAction(
+				'addExternalFeed',
+				new AjaxModal(
+						$router->url($request, null, null, 'addExternalFeed'),
+						__('plugins.generic.externalFeed.manager.create'),
+						'modal_add_item'
+						),
+				__('plugins.generic.externalFeed.manager.create'),
+				'add_item'
+			)
+		);
+		
+		// Columns
+		$cellProvider = new ExternalFeedGridCellProvider();
+		$this->addColumn(new GridColumn(
+			'title',
+			'plugins.generic.externalFeed.manager.title',
+			null,
+			'controllers/grid/gridCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'homepage',
+			'plugins.generic.externalFeed.manager.displayHomepage',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'displayBlockAll',
+			'plugins.generic.externalFeed.manager.displayBlockAll',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl', 
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'displayBlockHomepage',
+			'plugins.generic.externalFeed.manager.displayBlockHomepage',
+			null,
+			'controllers/grid/common/cell/selectStatusCell.tpl',
+			$cellProvider
+		));
+		
+	}
+	
+	/**
+	 * @copydoc GridHandler::initFeatures()
+	 */
+	function initFeatures($request, $args) {
+		import('lib.pkp.classes.controllers.grid.feature.OrderGridItemsFeature');
+		return array(new OrderGridItemsFeature());
+	}
+	
+	/**
+	 * @copydoc GridHandler::getDataElementSequence()
+	 */
+	function getDataElementSequence($row) {
+		return $row->getSequence();
+	}
+	
+	/**
+	 * @copydoc GridHandler::setDataElementSequence()
+	 */
+	function setDataElementSequence($request, $rowId, $gridDataElement, $newSequence) {
+		$externalFeedId = $rowId;
+		$journal = $request->getJournal();
+		$externalFeedDao = DAORegistry::getDAO('ExternalFeedDAO');
+		
+		// Ensure externalFeed is valid and for this journal
+		if (($externalFeedId != null && $externalFeedDao->getExternalFeedJournalId($externalFeedId) == $journal->getId())) {
+			$feed = $externalFeedDao->getExternalFeed($externalFeedId);
+			$feed->setSequence($newSequence);
+			$externalFeedDao->updateExternalFeed($feed);
+		}
+		
+	}
+	
+	//
+	// Overridden methods from GridHandler
+	//
+	/**
+	 * @copydoc Gridhandler::getRowInstance()
+	 */
+	function getRowInstance() {
+		return new ExternalFeedGridRow();
+	}
+	
+	//
+	// Public Grid Actions
+	//
+	/**
+	 * Display the grid's containing page.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 */
+	function index($args, $request) {
+		$context = $request->getContext();
+		import('lib.pkp.classes.form.Form');
+		$form = new Form(self::$plugin->getTemplatePath() . '/externalFeeds.tpl');
+		$json = new JSONMessage(true, $form->fetch($request));
+		return $json->getString();

```

Comment:
OTOH it's now possible to return a JSONMessage directly from a handler function -- no need to convert to string.